### PR TITLE
Add more response changelog checkers

### PR DESCRIPTION
**Pull request by blva**
- Add tests for response-property-became-optional checker
- Add response-became-optional checker

### CODE REVIEW

File: checker/check-response-property-became-required.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+				for _, mediaTypeDiff := range modifiedMediaTypes {
+					if mediaTypeDiff.SchemaDiff == nil {
+						continue
+					}
+
+					if mediaTypeDiff.SchemaDiff.RequiredDiff != nil {
+						for _, changedRequiredPropertyName := range mediaTypeDiff.SchemaDiff.RequiredDiff.Added {
+							id := ResponsePropertyBecameRequiredCheckId
+							if mediaTypeDiff.SchemaDiff.Revision.Value.Properties[changedRequiredPropertyName] == nil {
+								// removed properties processed by the ResponseRequiredPropertyRemovedCheck check
```
Comment (reuvenharrison) [inline checker/check-response-property-became-required.go]: Please update this comment

File: checker/check-response-property-became-required.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+							for _, changedRequiredPropertyName := range requiredDiff.Added {
+								id := ResponsePropertyBecameRequiredCheckId
+								if propertyDiff.Base.Value.Properties[changedRequiredPropertyName] == nil {
+									continue
+								}
+								if propertyDiff.Base.Value.Properties[changedRequiredPropertyName].Value.WriteOnly {
+									id = ResponseWriteOnlyPropertyBecameRequiredCheck
+								}
+								if propertyDiff.Revision.Value.Properties[changedRequiredPropertyName] == nil {
+									// removed properties processed by the ResponseRequiredPropertyRemovedCheck check
```
Comment (reuvenharrison) [inline checker/check-response-property-became-required.go]: Please update this comment

Comment (blva) [reply checker/check-response-property-became-required.go]: thanks! updated

File: checker/check-response-property-became-required.go (hunk 1 above)
Comment (blva) [reply checker/check-response-property-became-required.go]: thanks! updated

**Pull request merged**